### DISCUSSION TITLE
[Feature Request] Follow XDG Base Directory specification

### DISCUSSION THREAD

**Discussion by treierxyz** [TeamRizu/OutFox#262] [03 - Feature Requests]
Having hundreds of dot-files/folders in my home directory already, it's not great that Stepmania makes another one for it's configs and data. If there's a seperate folder for 5.3, then it would be nice if it adhered to XDG Base Directory specifications.

For example, data files could go into $XDG_DATA_HOME/stepmania-5.3, but if $XDG_DATA_HOME is not set, it's up to the game to default to $HOME/.local/share/stepmania-5.3, and as a last resort use $HOME/.stepmania-5.3. It might be easiest to just move everything into $XDG_DATA_HOME only and not use $XDG_CONFIG_HOME or the others (exept for $XDG_CACHE_HOME, which shouldn't be any harder to implement), as it seems seperating stuff *might* cause more trouble than it would help.

There are some issues on stepmania/stepmania about this as well:
https://github.com/stepmania/stepmania/issues/299
https://github.com/stepmania/stepmania/issues/1912

I'm not an expert on this, but it would be a nice addition, although it might be a pain in the butt to implement.

Here are some resources to follow:
https://wiki.archlinux.org/index.php/XDG_Base_Directory
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Thank you for your time.

**Comment by Scraticus** [comment]
Yep, this has been long overdue. I'll work on it and get it sorted for Alpha 5.x, likely in August/September.
